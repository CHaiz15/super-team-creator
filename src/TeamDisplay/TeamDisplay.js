import React from 'react'
import './TeamDisplay.scss'
import { connect } from 'react-redux'
import { Loader } from '../Loader/Loader'
import { addFavorite } from '../Actions'

export const TeamDisplay = ({team, name, memberOne, memberTwo, memberThree, addFavorite}) => {
  return ( 
    !memberOne ? <Loader />:
      <>
        <h1 className='creation-title'>{name}</h1>
        <button type='button' className='build-btn' onClick={() => addFavorite(team)}>Add to Favorites</button>
        <section className='team-container'>
          <div>
            <h3>{memberOne.name}</h3>
            <img className='member-image' src={memberOne.image.url} alt='member one'/>
            <ul>Stats:
              <li>Intelligence: {memberOne.powerstats.intelligence}</li>
              <li>strength: {memberOne.powerstats.strength}</li>
              <li>Speed: {memberOne.powerstats.speed}</li>
              <li>Durability: {memberOne.powerstats.durability}</li>
              <li>Power: {memberOne.powerstats.power}</li>
              <li>Combat: {memberOne.powerstats.combat}</li>
            </ul>
          </div>
          <div>
            <h3>{memberTwo.name}</h3>
            <img className='member-image' src={memberTwo.image.url} alt='member two'/>
            <ul>Stats:
              <li>Intelligence: {memberTwo.powerstats.intelligence}</li>
              <li>strength: {memberTwo.powerstats.strength}</li>
              <li>Speed: {memberTwo.powerstats.speed}</li>
              <li>Durability: {memberTwo.powerstats.durability}</li>
              <li>Power: {memberTwo.powerstats.power}</li>
              <li>Combat: {memberTwo.powerstats.combat}</li>
            </ul>
          </div>
          <div>
            <h3>{memberThree.name}</h3>
            <img className='member-image' src={memberThree.image.url} alt='member three'/>
            <ul>Stats:
              <li>Intelligence: {memberThree.powerstats.intelligence}</li>
              <li>strength: {memberThree.powerstats.strength}</li>
              <li>Speed: {memberThree.powerstats.speed}</li>
              <li>Durability: {memberThree.powerstats.durability}</li>
              <li>Power: {memberThree.powerstats.power}</li>
              <li>Combat: {memberThree.powerstats.combat}</li>
            </ul>
          </div>
          <div>
            <h3>Team Stats:</h3>
            <ul>Stats:
              <li>Intelligence: {parseInt(memberOne.powerstats.intelligence) + parseInt(memberTwo.powerstats.intelligence) + parseInt(memberThree.powerstats.intelligence)}</li>
              <li>strength: {parseInt(memberOne.powerstats.strength) + parseInt(memberTwo.powerstats.strength) + parseInt(memberThree.powerstats.strength)}</li>
              <li>Speed: {parseInt(memberOne.powerstats.speed) + parseInt(memberTwo.powerstats.speed) + parseInt(memberThree.powerstats.speed)}</li>
              <li>Durability: {parseInt(memberOne.powerstats.durability) + parseInt(memberTwo.powerstats.durability) + parseInt(memberThree.powerstats.durability)}</li>
              <li>Power: {parseInt(memberOne.powerstats.power) + parseInt(memberTwo.powerstats.power) + parseInt(memberThree.powerstats.power)}</li>
              <li>Combat: {parseInt(memberOne.powerstats.combat) + parseInt(memberTwo.powerstats.combat) + parseInt(memberThree.powerstats.combat)}</li>
            </ul>
          </div>
        </section>
      </>
    )
  }


export const mapStateToProps = state => ({
  team: state.team,
  name: state.team.name,
  memberOne: state.team.memberOne,
  memberTwo: state.team.memberTwo,
  memberThree: state.team.memberThree,
})

export const mapDispatchToProps = dispatch => ({
  addFavorite: team => dispatch(addFavorite(team)),
})

export default connect(mapStateToProps, mapDispatchToProps)(TeamDisplay)