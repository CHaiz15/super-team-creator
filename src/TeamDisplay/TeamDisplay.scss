@import '../variables';

body {
  overflow: hidden;
}

.screen-top {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  width: 100%;
  height: 10vh;
  margin: -10% 0 0 0;
}

.team-title {
  @include mainTitles();
  margin: 4% 0 0 0;
}

.whack-fav-btn {
  padding-top: 20px;
  font-family: 'Bangers';
  background-color: white;
  width: 11rem;
  height: 3rem;
  outline: none;
  border-radius: 20px;
  font-size: 1.5em; 
  margin: 18% 0 0 0;
  cursor: pointer;
  box-shadow: 0 3px black;
  animation: slide-top 0.5s cubic-bezier(0.250, 0.460, 0.450, 0.940) both;
}

.whack-fav-btn:hover {
  @include mainBtnHover();
}

.whack-fav-btn:active {
  @include mainBtnActive();
}


.fav-btn {
  font-family: 'Bangers';
  background-color: white;
  width: 11rem;
  height: 4rem;
  outline: none;
  border-radius: 20px;
  font-size: 1.5em; 
  margin: 18% 0 0 0;
  cursor: pointer;
  box-shadow: 0 3px black;
  animation: slide-top 0.5s cubic-bezier(0.250, 0.460, 0.450, 0.940) both;
}

.fav-btn:hover {
  @include mainBtnHover();
}

.fav-btn:active {
  @include mainBtnActive();
}

.team-container {
  font-size: 1.2em;
  margin: 13% 0 0 0;
  color: black;
  height: 111vh;
}

.members-container {
  height: 50%;
  width: 100%;
  display: flex;
  justify-content: center;
  flex-direction: row;
}

li, ul {
  list-style: none;
  padding: 0;
}

.member-div {
  background-color: white;
  opacity: 0.95;
  border-radius: 10px;
  margin: 0 1% 0 1%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
}

.member-div:hover {
  background-color: $primary-red;
  cursor: pointer;
}

.member-image {
  align-self: center;
  border-radius: 50%;
  height: 35%;
  width: 35%;
}

.team-stats {
  background-color: black;
  opacity: 0.9;
  color: white;
  text-align: center;
  display: flex;
  justify-content: center;
  width: 100%;
  margin: 2% 0 0 0;
}

