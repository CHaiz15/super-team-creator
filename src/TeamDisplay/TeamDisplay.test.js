import React from 'react';
import { TeamDisplay } from './TeamDisplay.js';
import { shallow } from 'enzyme'

describe('TeamDisplay', () => {
  let wrapper;
  let mockTeam;
  beforeEach(() => {
      mockTeam = {
        name:'Best Team Ever',
        memberOne: {
          name:'Hulk', 
          powerstats: {
            Intelligence: 81,
            strength: 40,
            Speed: 29,
            Durability: 55,
            Power: 63,
            Combat: 90,
          },
          image: {url: "https://www.superherodb.com/pictures2/portraits/10/100/83.jpg"},
        },
        memberTwo: {
          name:'Thanos',
          powerstats: {
            Intelligence: 81,
            strength: 40,
            Speed: 29,
            Durability: 55,
            Power: 63,
            Combat: 90,
          },
          image: {url: "https://www.superherodb.com/pictures2/portraits/10/100/83.jpg"},
        },
        memberThree: {
          name:'Batman',
          powerstats: {
            Intelligence: 81,
            strength: 40,
            Speed: 29,
            Durability: 55,
            Power: 63,
            Combat: 90,
          },
          image: {url: "https://www.superherodb.com/pictures2/portraits/10/100/83.jpg"},
        },
        id: 123,
      }
    wrapper = shallow(<TeamDisplay team={mockTeam} name={mockTeam} memberOne={mockTeam.memberOne} memberTwo={mockTeam.memberTwo} memberThree={mockTeam.memberThree} addFavorite={jest.fn()} />);
  })  
  it('should be an instance of the TeamDisplay component', () => {
      expect(wrapper).toMatchSnapshot();
  });
})
