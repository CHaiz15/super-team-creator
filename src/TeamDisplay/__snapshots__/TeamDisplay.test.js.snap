// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamDisplay should be an instance of the TeamDisplay component 1`] = `ShallowWrapper {}`;
